packages/core/src/actions/assign.ts

```diff
@@ -28,71 +85,17 @@ export function assign<
   assignment:
     | Assigner<LowInfer<TContext>, TExpressionEvent>
     | PropertyAssigner<LowInfer<TContext>, TExpressionEvent>
-): DynamicAssignAction<TContext, TExpressionEvent, TEvent> {
-  return createDynamicAction<
-    TContext,
-    TExpressionEvent,
-    TEvent,
-    AssignActionObject<TContext>,
-    {
-      assignment: typeof assignment;
+) {
+  function assign(_: ActionArgs<TContext, TExpressionEvent>) {
```

Comment:
Nit: was a little confused, can we call this something different than `assign` to disambiguate?

Reply:
this was somewhat deliberate here - a) naming is hard b) it will show be printed in the console as `ƒ assign() {}` which is kinda nice. I agree that this is a little bit confusing internally but I just don't know what is a better name for this ;p

Reply:
All good 👍